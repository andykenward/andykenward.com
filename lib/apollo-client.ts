import { InMemoryCache, NormalizedCacheObject } from "apollo-cache-inmemory"
import { ApolloClient } from "apollo-client"
import { HttpLink } from "apollo-link-http"
import fetch from "isomorphic-unfetch"
import { NextPageContext } from "next"

export default function createApolloClient(
  initialState: NormalizedCacheObject = {},
  ctx?: NextPageContext
) {
  // The `ctx` (NextPageContext) will only be present on the server.
  // use it to extract auth headers (ctx.req) or similar.
  return new ApolloClient({
    ssrMode: Boolean(ctx),
    link: new HttpLink({
      uri: `${process.env.ENDPOINT}`, // Server URL (must be absolute)
      credentials: "same-origin", // Additional fetch() options like `credentials` or `headers`
      fetch,
    }),
    cache: new InMemoryCache().restore(initialState),
  })
}
