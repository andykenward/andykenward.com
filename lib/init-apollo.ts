import {
  ApolloClient,
  InMemoryCache,
  HttpLink,
  NormalizedCacheObject,
} from "apollo-boost"
import isoFetch from "isomorphic-unfetch"

let apolloClient: ApolloClient<NormalizedCacheObject> | null = null

function create(initialState: NormalizedCacheObject | undefined) {
  // Check out https://github.com/zeit/next.js/pull/4611 if you want to use the AWSAppSyncClient
  const isBrowser: boolean = typeof window !== "undefined"

  return new ApolloClient({
    connectToDevTools: isBrowser,
    ssrMode: !isBrowser, // Disables forceFetch on the server (so queries are only run once)
    link: new HttpLink({
      uri:
        "https://api-euwest.graphcms.com/v1/cjywu05hg0ysg01dj8nkna8ma/master", // Server URL (must be absolute)
      credentials: "same-origin", // Additional fetch() options like `credentials` or `headers`
      // Use fetch() polyfill on the server
      fetch: !isBrowser ? isoFetch : fetch,
    }),
    cache: new InMemoryCache().restore(initialState || {}),
  })
}

export default function initApollo(
  initialState?: NormalizedCacheObject | undefined
) {
  // Make sure to create a new client for every server-side request so that data
  // isn't shared between connections (which would be bad)
  if (typeof window === "undefined") {
    return create(initialState)
  }

  // Reuse client on the client-side
  if (!apolloClient) {
    apolloClient = create(initialState)
  }

  return apolloClient
}
