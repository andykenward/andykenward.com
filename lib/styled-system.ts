import {
  CSSObject,
  InterpolationWithTheme,
  SerializedStyles,
} from "@emotion/core"
import * as CSS from "csstype"
import { is } from "ramda"
import {
  compose,
  fontFamily,
  fontStyle,
  fontWeight,
  letterSpacing,
  lineHeight,
  style,
  textAlign,
} from "styled-system"

/**
 * INTERFACES
 */
export interface StyledProps {
  as?: keyof JSX.IntrinsicElements | React.ComponentType<any> // eslint-disable-line
  css?: CSSObject | SerializedStyles | InterpolationWithTheme<any> // eslint-disable-line @typescript-eslint/no-explicit-any
}

export interface TextTransformProps {
  textTransform?: CSS.TextTransformProperty
}

/**
 *
 * UTILS
 */
export const fontStack = (fonts: string[]): string =>
  fonts.map(font => (font.includes(" ") ? `"${font}"` : font)).join(", ")

export const convertToRem = (n: number | string): string | number =>
  is(Number, n) ? `${(n as number) / 16}rem` : n

export const propKeys = ["color", "fontSize"]

/**
 * CUSTOM styles
 */
const textTransform = style({
  prop: "textTransform",
  cssProperty: "textTransform",
})

export const fontSize = style({
  prop: "fontSize",
  cssProperty: "fontSize",
  transformValue: convertToRem,
})

/**
 * COMPOSED STYLES
 */

export const typography = compose(
  fontFamily,
  // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
  //@ts-ignore
  fontSize,
  fontWeight,
  lineHeight,
  letterSpacing,
  textAlign,
  fontStyle,
  textTransform
)
