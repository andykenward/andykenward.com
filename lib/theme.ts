import * as CSS from "csstype"
import { is } from "ramda"
import {
  compose,
  fontFamily,
  fontStyle,
  fontWeight,
  letterSpacing,
  lineHeight,
  style,
  system,
  textAlign,
} from "styled-system"

const fontStack = (fonts: string[]): string =>
  fonts.map(font => (font.includes(" ") ? `"${font}"` : font)).join(", ")

export const convertToRem = (n: number | string): string | number =>
  is(Number, n) ? `${(n as number) / 16}rem` : n

const fontSizes = [14, 16, 20, 28, 36, 38, 40, 44, 46, 48, 50, 54, 64, 80]

const textTransform = style({
  prop: "textTransform",
  cssProperty: "textTransform",
})

export interface TextTransform {
  textTransform?: CSS.TextTransformProperty
}

const customFontSize = system({
  fontSize: {
    property: "fontSize",
    scale: "fontSizes",
    transform: convertToRem,
  },
  ...textTransform,
})

export const typography = compose(
  fontFamily,
  customFontSize,
  fontWeight,
  lineHeight,
  letterSpacing,
  textAlign,
  fontStyle
)

const fonts = {
  regular: fontStack([
    "Graphik",
    "Helvetica Neue",
    "Helvetica",
    "Arial",
    "Verdana",
    "sans-serif",
  ]),
  brand: fontStack([
    "Cooper Light",
    "Georgia",
    "Times",
    "Times New Roman",
    "serif",
  ]),
} as const

// : { [key: string]: CSS.FontWeightProperty }
const fontWeights = {
  light: 300,
  regular: 400,
  medium: 500,
} as const

type FontWeightKeys = keyof typeof fontWeights

const lineHeights = {
  condensed: 1.25,
  default: 1.5,
} as const

type LineHeightKeys = keyof typeof lineHeights

const space = {
  0: 0,
  4: convertToRem(4),
  8: convertToRem(8),
  16: convertToRem(16),
  32: convertToRem(32),
  56: convertToRem(56),
  64: convertToRem(64),
  128: convertToRem(128),
} as const

export const theme = {
  fonts: fonts,
  fontWeights: fontWeights,
  fontSizes: fontSizes,
  lineHeights: lineHeights,
  space: space,
} as const
