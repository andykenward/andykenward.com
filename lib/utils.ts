import shouldForwardProp from "@styled-system/should-forward-prop"
import { equals, indexOf } from "ramda"

import { propKeys } from "./styled-system"

export const filterProps: (prop: string) => boolean = (prop: string): boolean =>
  shouldForwardProp(prop) && equals(indexOf(prop, propKeys), -1)

export const childrenOnly: (prop: string) => boolean = (
  prop: string
): boolean => prop === "children" ?? prop === "data-test-id"
