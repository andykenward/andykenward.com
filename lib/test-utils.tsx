import {
  MockedProvider,
  MockedResponse,
  ResultFunction,
} from "@apollo/react-testing"
import { render } from "@testing-library/react"
import {
  DocumentNode,
  FetchResult,
  InMemoryCache,
  IntrospectionFragmentMatcher,
} from "apollo-boost"
import { ThemeProvider } from "emotion-theming"

import result from "../generated/introspection-result"
import { theme } from "./theme"

export const renderWithTheme = (element: React.ReactElement<any>) =>
  render(<ThemeProvider theme={theme}>{element}</ThemeProvider>)

const fragmentMatcher = new IntrospectionFragmentMatcher({
  introspectionQueryResultData: result,
})

export const renderWithApp = (
  element: React.ReactElement,
  MOCKS: MockedResponse[] = []
) => {
  const cache = new InMemoryCache({ fragmentMatcher })
  return render(
    <MockedProvider mocks={MOCKS} cache={cache}>
      <ThemeProvider theme={theme}>{element}</ThemeProvider>
    </MockedProvider>
  )
}

// TODO: fix response typings
export const getMock = (
  query: DocumentNode,
  response?: unknown,
  error = false
): MockedResponse[] => [
  {
    request: { query: query },
    result: response
      ? ({ data: response } as FetchResult | ResultFunction<FetchResult>)
      : undefined,
    error: error ? new Error("error from server") : undefined,
  },
]

export const getMocksError = (query: DocumentNode) =>
  getMock(query, undefined, true)
