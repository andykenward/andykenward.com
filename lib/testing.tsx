import { render } from "@testing-library/react"
import { ThemeProvider } from "emotion-theming"

import { theme } from "./theme"

export const renderWithTheme = (element: React.ReactElement<any>) =>
  render(<ThemeProvider theme={theme}>{element}</ThemeProvider>)
