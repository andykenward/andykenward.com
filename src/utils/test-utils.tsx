import { render, RenderOptions, RenderResult } from "@testing-library/react"
import { ThemeProvider } from "theme-ui"

import { theme } from "../theme"

const AllTheProviders: React.FC = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>
}

const customRender = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult => render(ui, { wrapper: AllTheProviders, ...options })

export * from "@testing-library/react"

export { customRender as render }
