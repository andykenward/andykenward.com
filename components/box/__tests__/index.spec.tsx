import { Box } from ".."
import { renderWithTheme } from "../../../lib/test-utils"

describe("<Box />", () => {
  it("renders correctly", () => {
    const { asFragment } = renderWithTheme(<Box />)
    expect(asFragment()).toMatchSnapshot()
  })
})
