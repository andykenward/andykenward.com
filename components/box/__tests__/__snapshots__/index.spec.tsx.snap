// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Box /> renders correctly 1`] = `
<DocumentFragment>
  <div
    class="css-3emocq-Box emotion-0"
  />
</DocumentFragment>
`;
